syntax = "proto3";
package eos.ns;

//------------------------------------------------------------------------------
// Container metadata protocol buffer object
//------------------------------------------------------------------------------
message ContainerMdProto {
  uint64 id = 1;
  uint64 parent_id = 2;
  int64 uid = 3;
  int64 gid = 4;
  int64 tree_size = 6;
  int32 mode = 5;
  uint32 flags = 7;
  uint32 acl_id = 8;
  string name = 9;
  bytes ctime = 10; // change time
  bytes mtime = 11; // modification time
  bytes stime = 12; // sync time
  map<string, string> xattrs = 13;
}
